src/crypto/psa.c

```diff
@@ -1,4 +1,14 @@
-#include <psa/crypto.h>
+/* Arduino mbed architectures use mbedtls crypto but include
+ * all source files in compilation, so add a guard
+ */
+#if !defined(ARDUINO_ARCH_MBED)
```

Comment:
Same thing ... this guard should be solved by the build system. We are mixing logics imho